woocommerce/woocommerce-ios
# Store creation M3: profiler question - optional selling platforms

Part of #8377 


## Description


This PR only includes the UI part, as the API is still WIP (p1670825052838929/1670823717.177509-slack-C01SFMVEYAK). The profiler question is only shown behind a new feature flag `storeCreationM3Profiler`, and the selected option isn't shown in the store summary screen so the value doesn't need to be passed to subsequent view controllers.

After the main question about the selling status in https://github.com/woocommerce/woocommerce-ios/pull/8481, this PR implemented the follow-up question on the eCommerce platforms for merchants who indicate that they're already selling online. A new SwiftUI `StoreCreationSellingPlatformsQuestionView` was created for the platforms question. The first selling status question view `StoreCreationSellingStatusQuestionView` was refactored to just contain the selling status question content, and a new container view `StoreCreationSellingStatusQuestionContainerView` was created to show either the platforms question or the selling status question based on the view model's `isAlreadySellingOnline` boolean.

#### Another navigation implementation I tried

First I tried navigating to the platforms question view from the main selling status question view using `LazyNavigationLink`/`NavigationLink`, but a new issue on the navigation bar background appeared. The background color of the navigation bar is set to the default color instead of the custom transparent color set in the hosting controller. The API to [configure the navigation bar color in SwiftUI](https://developer.apple.com/documentation/swiftui/view/toolbarbackground(_:for:)-5ybst) is only available in iOS 16+, while our app still supports iOS 15.

## Testing instructions


- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the store name form should be shown
- Enter a store name and continue
- On the store category screen, choose an option and continue or skip --> the selling status profiler question should be shown
- Tap `Skip` --> the domain selector should be shown
- Go back to the previous screen
- Select an option that's not "I'm already selling online" and tap `Continue` --> the domain selector should be shown
- Go back to the previous screen
- Select the option "I'm already selling online" --> a list of eCommerce platforms should be shown for selection
- Select one or multiple platforms --> the platforms should be in the selected state
- Select the same platform again --> the platform should be deselected
- Tap `Continue` --> the domain selector should be shown

## Screenshots


dark | light
-- | --
![Simulator Screen Shot - iPhone 14 Pro - 2022-12-28 at 13 43 12](https://user-images.githubusercontent.com/1945542/209763789-e062b883-b1fd-400b-bf99-f821635bb077.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2022-12-28 at 13 43 33](https://user-images.githubusercontent.com/1945542/209763796-2073341b-a9bb-4943-85ba-3a21165e7d0c.png)


---
